python_modules/libraries/dagster-dbt/dagster_dbt_tests/rpc/test_resources.py

```diff
@@ -111,3 +118,123 @@ def a_solid(context):
 
     execute_solid(a_solid, ModeDefinition(resource_defs={"dbt_rpc": local_dbt_rpc_resource}))
     assert it["ran"]
+
+
+def test_dbt_rpc_sync_resource():
+    it = {}
+
+    @solid(required_resource_keys={"dbt_rpc"})
+    def a_solid(context):
+        assert isinstance(context.resources.dbt_rpc, DbtRpcSyncClient)
+        assert context.resources.dbt_rpc.host == "<default host>"
+        assert context.resources.dbt_rpc.port == 8580
+        it["ran"] = True
+
+    execute_solid(
+        a_solid,
+        ModeDefinition(resource_defs={"dbt_rpc": dbt_rpc_sync_resource}),
+        None,
+        None,
+        {"resources": {"dbt_rpc": {"config": {"host": "<default host>"}}}},
+    )
+    assert it["ran"]
+
+
+@pytest.mark.parametrize(
+    "client_class,resource",
+    [(DbtRpcClient, dbt_rpc_resource), (DbtRpcSyncClient, dbt_rpc_sync_resource)],
+)
+def test_dbt_rpc_resource_status(dbt_rpc_server, client_class, resource):
+    @solid(required_resource_keys={"dbt_rpc"})
+    def compile_solid(context):
+        assert isinstance(context.resources.dbt_rpc, client_class)
+        out = context.resources.dbt_rpc.status()
+        return out
+
+    result = execute_solid(
+        compile_solid,
+        ModeDefinition(resource_defs={"dbt_rpc": resource.configured({"host": "localhost"})}),
+    )
+
+    assert result.success
+    assert isinstance(result.output_value("result"), DbtRpcOutput)
+
+
+@pytest.mark.parametrize(
```

Comment:
these tests look good, but I would like to see at least one or two tests that make sure that the sync client is actually waiting for the process to finish.

one way to do this would be by inspecting the result object after a run() command. if it's async (not waiting), then the DbtRpcOutput (specifically the `result` property of that object) will look different than in the synchronous case. Try running it both ways (while printing out the entire DbtRpcOutput object) and you'll see what I mean.

Reply:
addressed, added 2 tests that test that the async response contains a request token and that the sync response contains elapsed time.